### ISSUE TITLE
LDAP authentication bypass

### ISSUE THREAD

**Issue by moonwolf-github** [gitbucket/gitbucket#2487]
With LDAP authentication it is possible to access repositories through http on behalf any user (giving he has acces to that repository) by passing no password at all. I'm not entirely sure yet it's gitbucket issue or maybe my LDAP server configuration problem. Note that one cannot login that way as on login page there is no way to post empty password.

**Comment by moonwolf-github** [comment]
I don't know if #1903 is related.

**Comment by moonwolf-github** [comment]
I did test regarding our LDAP server and it looks OK i guess:
```
[root@vader ~]# ldapwhoami -x -D uid=moonwolf,ou=People,ou=main,o=tech  -W
Enter LDAP Password: [just ENTER]
ldap_bind: Server is unwilling to perform (53)
        additional info: Unauthenticated binds are not allowed
[root@vader ~]# ldapwhoami -x -D uid=moonwolf,ou=People,ou=main,o=tech  -W
Enter LDAP Password: [valid password]
dn: uid=moonwolf,ou=people,ou=main,o=tech
```

**Comment by takezoe** [comment]
@moonwolf-github The root cause is fallback mechanism when LDAP authentication is enabled. This issue is fixed in #2488 and will be released in the next version. Great thanks for catching this!
